prog/weaver/main.go

```diff
@@ -368,6 +376,10 @@ func main() {
 		go listenAndServeHTTP(statusAddr, statusMux)
 	}
 
+	if enablePlugin {
+		go plugin.Start(httpAddr, dockerAPI, pluginSocket, pluginMeshSocket)
```

Comment:
When initializing `go-dockerclient`, the initialization routine `NewClient` sends the `/version` request. If Docker is not listening on the socket, the request will fail, the weaver process will crash (due to `log.Fatal`), and weaver will get restarted by Docker.